### PR TITLE
Import secret and access secret resources

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description

Implement ImportState fully for both the secret and access secret resources. 

## Type of change

- Update to resources not yet released.

## Environment

- Juju controller version: 3.5-beta1

- Terraform version: 1.8.1

## QA steps

```tf
terraform {
  required_providers {
    juju = {
      version = ">= 0.12.0"
      source  = "juju/juju"
    }
  }
}

provider "juju" {}

resource "juju_model" "my_model" {
   name = "tf-secret-test"
}

/*
resource "juju_secret" "my_secret" {
  model = juju_model.my_model.name
  name  = "my_secret_name"
  value = {
    key1 = "value1"
    key2 = "value2"
  }
  info  = "This is my secret"
}

resource "juju_access_secret"  "my_secret_access" {
  model = juju_model.my_model.name
  applications = [
    juju_application.jul.name, juju_application.jul2.name
  ]
  secret_id = juju_secret.my_secret.secret_id
}
*/

resource "juju_application" "jul" {
  name  = "jul"
  model = juju_model.my_model.name

  charm {
    name     = "jameinel-ubuntu-lite"
    channel  = "latest/stable"
  }

  units = 1
}

resource "juju_application" "jul2" {
  name  = "jul2"
  model = juju_model.my_model.name

  charm {
    name     = "jameinel-ubuntu-lite"
    channel  = "latest/stable"
  }

  units = 1
}
```

```
# Using the plan above, run:
$ terraform init && terraform plan && terraform apply -auto-approve
# Only the model and two applications will be created

# Create the secret and access details in juju
$ juju add-secret my_secret_name --info "This is my secret" key1=value1 key2=value2
secret:cojb5haeidiish9hvcrg
$ juju grant-secret my_secret_name jul,juj2

# Import the two new import resources
$ terraform import juju_secret.my_secret tf-secret-test:my_secret_name
$ terraform import juju_access_secret.my_secret_access tf-secret-test:my_secret_name

# Edit the plan: copy the contents of my_secret to my_secret_import and the contents of my_access_secret to my_access_secret_import
# Run 
$ terraform plan
# No changes should be necessary.
```

## Additional notes

JUJU-5904

### CODE REVIEW

File: internal/provider/resource_access_secret.go [hunk 1]
```diff
@@ -43,10 +43,59 @@ type accessSecretResourceModel struct {
...
+			"Unexpected Import Identifier",
+			fmt.Sprintf("Expected import identifier with format: <modelname>:<secretname>. Got: %q", req.ID),
+		)
+		return
+	}
+	modelName := parts[0]
+	secretName := parts[1]
+
+	// Prevent panic if the provider has not been configured.
+	if s.client == nil {
```
Comment (cderici) [inline internal/provider/resource_access_secret.go:66]: move me at the beginning of the function, no need to do strings.Split if the client is nil.

File: internal/provider/resource_secret.go [hunk 2]
```diff
@@ -45,10 +46,61 @@ type secretResourceModel struct {
...
+	}
+	modelName := parts[0]
+	secretName := parts[1]
+
+	// Prevent panic if the provider has not been configured.
+	if s.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, "secret", "import")
+		return
+	}
+	readSecretOutput, err := s.client.Secrets.ReadSecret(&juju.ReadSecretInput{
```
[the remark this answers is not in the document]
Comment (hmlanigan) [reply internal/provider/resource_secret.go:73]: It's at the bottom of the method per usual

File: internal/provider/resource_access_secret.go (hunk 1 above)
Comment (hmlanigan) [reply internal/provider/resource_access_secret.go:66]: it's a vice versa thing to me. either could fail. moving

File: internal/provider/resource_secret.go [hunk 3]
```diff
@@ -45,10 +46,61 @@ type secretResourceModel struct {
...
+			"Unexpected Import Identifier",
+			fmt.Sprintf("Expected import identifier with format: <modelname>:<secretname>. Got: %q", req.ID),
+		)
+		return
+	}
+	modelName := parts[0]
+	secretName := parts[1]
+
+	// Prevent panic if the provider has not been configured.
+	if s.client == nil {
```
[the remark this answers is not in the document]
Comment (hmlanigan) [reply internal/provider/resource_secret.go:69]: it's a vice versa thing to me. either could fail. moving

**Review by anvial — APPROVED** [review]
LGTM! Thx, passed well.

PS: Thx for fixing (ID issue) in tests.

**Pull request merged**